test/test_helper.rb

```diff
@@ -1,5 +1,12 @@
 require "simplecov"
-SimpleCov.start "rails"
+SimpleCov.start "rails" do
+  enable_coverage :branch
+  minimum_coverage 95
+  add_filter "/lib/api_error_routing_constraint.rb"
+  add_filter "/lib/bank_holiday_generator.rb"
+  add_filter "/lib/format_routing_constraint.rb"
+  add_filter "/lib/frontend.rb"
```

Comment:
Why are we not checking coverage on these files? 

We've got a test https://github.com/alphagov/frontend/blob/677081b87c10bd49d7d57e9695d7f78553a3c140/test/unit/format_routing_constraint_test.rb#L3 - is not covering everything?

Reply:
We should add tests I think rather than skip checking them when we calculate coverage

Reply:
Yes, you're right, but those files were excluded because simplecov doesn't register they're test-covered (they are). I suppose that's because they're loaded before the testing framework by Rails. After we move to RSpec (I think the other half or the web team is in the progress of doing that), those can be included again.

The routing constraint files are loaded by `config/routes.rb` and tested by `format_routing_constraint_test.rb` and `api_error_routing_constraint_test.rb`.
`frontend.rb` is loaded by some initializers in config e.g. `config/initializers/location_api.rb`. There is not specific test, but the file only adds accessors, so it's covered by the initializers (Rails won't start properly, if those don't work).
 `bank_holiday_generator.rb` is only used in a rake task and tested by `bank_holiday_generator_test.rb`.

![coverage](https://github.com/alphagov/frontend/assets/893013/f2171720-c035-4294-8fee-e32b1c2d3d3a)

Reply:
Ah nice ok. So let's make a card to update the simple cov config. The card will have a dependency on Frontend running RSpec https://github.com/alphagov/frontend/pull/4090

Reply:
I've added a Trello card for that: https://trello.com/c/WE1eBmlf/1079-reenable-test-coverage-checking-for-excluded-files-after-migrating-to-rspec